### PR TITLE
docs: update Node.js example to run as expected

### PR DESCRIPTION
**Pull request by HipsterBrown**
I'm very excited to be able to use the Viam TypeScript SDK for my Node.js projects! Thanks for all the hard work to make this integration possible @edaniels and @njooma! :clap:

After running through the current documentation and example application, I ran into a few issues that I'm resolving with this PR.

### CODE REVIEW

File: Node.md [hunk 1]
```diff
@@ -1,10 +1,10 @@
-# Node & Viam's TypeScript SDK
+# Node.js & Viam's TypeScript SDK
```
Comment (HipsterBrown) [inline Node.md:1]: Using the official name "Node.js" for consistency across the documentation.

File: Node.md [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
 
-This document contains detailed instructions on including Viam in your Node project. For a runnable example, see the [examples directory](/examples/node/).
+This document contains detailed instructions on including Viam in your Node.js project. For a runnable example, see the [examples directory](/examples/node/).
 
 ## Requirements
 
-This document assumes you already have Node installed. If not, follow the [instructions](https://nodejs.org/en/learn/getting-started/how-to-install-nodejs) provided by Node.
+This document assumes you already have Node.js >= version 20 installed. If not, follow the [instructions](https://nodejs.org/en/learn/getting-started/how-to-install-nodejs) provided by Node.js.
```
Comment (HipsterBrown) [inline Node.md:7]: The `--env-file` flag was introduced in v20, so calling this out just in case some folks are running v18 which is still under long term support.

File: Node.md [hunk 3]
```diff
@@ -19,25 +19,25 @@ You can use either Yarn or NPM to install the dependencies. This document will u
...
 ### Polyfills
 
-Using the SDK with Node also requires the use of some polyfills. In your application's entrypoint (`main.ts`, `index.ts`, or something similar), you will need to register those polyfills:
+Using the SDK with Node.js also requires the use of some polyfills. In your application's entrypoint (`main.ts`, `index.ts`, or something similar), you will need to register those polyfills:
 
 ```ts
 // main.ts
 
-import wrtc = require('node-datachannel/polyfill');
+const wrtc = require('node-datachannel/polyfill');
```
Comment (HipsterBrown) [inline Node.md:27]: Node.js supports two module formats: CommonJS (`require`) and ESM (`import`). While they can be mixed together in some programs, the expected syntax for each format is different. If we wanted to use ESM import statements it would look a little like reversed Python:

```ts
import wrtc from 'node-datachannel/polyfill';
```

However after updating the example code to use ESM, it appears the Viam TypeScript SDK does not have an ESM compatible module export for Node.js at the moment. So I've updated the example code to use CommonJS for now.

File: examples/node/package.json [hunk 4]
```diff
@@ -17,7 +17,6 @@
     "ts-protoc-gen": "^0.14.0",
     "typescript": "^5.2.2"
   },
-  "type": "module",
```
Comment (HipsterBrown) [inline examples/node/package.json:20]: This field enforces ESM for Node.js projects, which is not currently possible with the TS SDK, so removing this to run the project as CommonJS.

File: examples/node/src/main.ts [hunk 5]
```diff
@@ -1,6 +1,8 @@
-import VIAM = require('@viamrobotics/sdk');
-import wrtc = require('node-datachannel/polyfill');
-import connectNode = require('@connectrpc/connect-node');
+const VIAM = require('@viamrobotics/sdk');
+const wrtc = require('node-datachannel/polyfill');
+const connectNode = require('@connectrpc/connect-node');
+
+// @ts-expect-error
```
Comment (HipsterBrown) [inline examples/node/src/main.ts:5]: This is required to pass the TypeScript compiler (`tsc`). If the project was updated to use something like [`tsx`](https://tsx.is/node-enhancement#swap-node-for-tsx), it could run in a single command: `tsx  --env-file=.env src/main.ts`

Otherwise, once Node.js v22 is more widely used, it could be run directly without the extra dependency using the [type stripping feature](https://nodejs.org/en/learn/typescript/run-natively): `node --experimental-strip-types --env-file=.env src/main.ts`

File: examples/node/src/main.ts [hunk 6]
```diff
@@ -12,7 +14,7 @@ for (const key in wrtc) {
 async function connect() {
   const host = process.env.HOST;
   const apiKeyId = process.env.API_KEY_ID;
-  const apiKeySecret = process.env.API_KEY_SECRET;
+  const apiKeySecret = process.env.API_KEY;
```
Comment (HipsterBrown) [inline examples/node/src/main.ts:17]: The README uses `API_KEY` in the `.env` file instead of `API_KEY_SECRET` and this naming is more consistent across the other SDK documentation.

File: Node.md (hunk 3 above)
Comment (edaniels) [reply Node.md:27]: This file works fine for me in Node 20 mod your API key secret fix

File: examples/node/package.json (hunk 4 above)
Comment (edaniels) [reply examples/node/package.json:20]: same for node 20

**Review by edaniels — COMMENTED** [review]
This all looks good to me other than a discussion for main.ts (mod api key secret fix). 20 is about a year old and 18 is EOL in 6 months. I think it'd be better to be on 20.

File: Node.md (hunk 3 above)
Comment (HipsterBrown) [reply Node.md:27]: `import wrtc = require('node-datachannel/polyfill')` works in Node 20? That's quite odd since it doesn't work in the REPL:

image

Could `tsc` be transpiling the code to be valid?

Comment (edaniels) [reply Node.md:27]: Yes tsc or tsx would transpile to js since node can't run typescript.

Comment (njooma) [reply Node.md:27]: I believe eslint/the community in general prefers ESM style imports in TS? At least i often see lint errors when i do `const foo = require('bar')`

Comment (HipsterBrown) [reply Node.md:27]: Node 22 can run TypeScript with the experimental type stripping flag and would fail with the example code as is. I think we should have valid JS in the examples, even if tsc papers over the syntax errors.

Comment (HipsterBrown) [reply Node.md:27]: > I believe eslint/the community in general prefers ESM style imports in TS? At least i often see lint errors when i do const foo = require('bar')

Yes, ESM is the official standard and should be used when possible. However, the current build output of the TS SDK cannot be imported as ESM in Node.js:

image

(The above example in the REPL is using the dynamic ESM import syntax but the same error applies when running `main.ts` with static ESM imports)

**Comment by HipsterBrown** [comment]
I don't know why the E2E tests are failing. I ran them locally with the same result.

**Comment by edaniels** [comment]
@HipsterBrown maybe it's best to move to tsx instead of tsc while getting `node --experimental-strip-types src/main.ts` to work too? Whatever gives the most compat I'm all for

File: e2e/tests/connect-and-status.spec.ts [hunk 7]
```diff
@@ -7,12 +7,12 @@ test('check resource names and multiple statuses', async ({ page }) => {
   const resourceNames = page
     .getByTestId('resource-names')
     .getByTestId('resource-name');
-  await expect(resourceNames).toHaveCount(4);
+  await expect(resourceNames).toHaveCount(3);
```
Comment (HipsterBrown) [inline e2e/tests/connect-and-status.spec.ts:10]: I'm not sure why the number of builtin resources is only 1 now, maybe something changed in the server implementation. It appears to be unrelated to my changes.

**Review by edaniels — APPROVED** [review]

**Review by njooma — APPROVED** [review]
